mapbook-iOS/Enums/AppMode.swift

```diff
@@ -22,8 +22,39 @@
 // email: contracts@esri.com
 //
 
-enum AppMode:String {
-    case notSet = "NotSet"
-    case device = "Device"
-    case portal = "Portal"
+enum AppMode: Int {
+    case portal, device
+    
+    private static let userDefaultsKey = "AppMode.UserDefaultsKey"
+    
+    func saveToUserDefaults() {
+        UserDefaults.standard.set(self.rawValue, forKey: AppMode.userDefaultsKey)
+    }
+    
+    static func retrieveFromUserDefaults() -> AppMode {
+        let rawAppMode = UserDefaults.standard.integer(forKey: userDefaultsKey)
+        return AppMode(rawValue: rawAppMode) ?? .portal
```

Comment:
This works, but I just want to make sure it works deliberately and not just by accident. If there's no `UserDefaults` value for app mode, the `rawAppMode` will come back as `0`, which will be parsed to `.portal`. So the nil-optional  coalescing operator here is defending against an unexpected value being stored in UserDefaults, and not no value.

Should you explicitly declare `case portal = 0, device`, to be more explicit, and in case Swift ever changes its behavior?

Reply:
how about this:
```swift
guard seanConnery != jamesBond else {
    fatalError("Don't watch!")
}
```
As for a few of your notes:

> I just want to make sure it works deliberately and not just by accident

It was designed deliberately though perhaps it tells a better story if we explicitly specify the case value.

> So the nil-optional coalescing operator here is defending against an unexpected value being stored in UserDefaults, and not no value.

Even if we specify the first case as 0, we will still need the nil-optional coalescing operator.

> and in case Swift ever changes its behavior

This feature is one of the early Swift hallmark features and there doesn't appear to be any plans to change it. If they do change it, then we will know and can account accordingly. 

At the end of it all, however, in the spirit of open source software storytelling, i'll take your suggestion.

Reply:
> Even if we specify the first case as 0, we will still need the nil-optional coalescing operator.

Oh, yeah, for sure.

I think the change makes sense not because I really expect Swift to change anything here in terms of enum indexing, but to be explicit about our expectations, in particular for those who don't know that UserDefaults returns `0` (perhaps it should really return `nil` if it were properly Swifty). Thanks for making the change.